### ISSUE TITLE
Move opera examples to integration test folder

### ISSUE THREAD

**Issue by anzoman** [xlab-si/xopera-opera#93]
## Description
Opera examples currently reside in [`/examples`](https://github.com/xlab-si/xopera-opera/tree/master/examples) folder, whereas opera runnable integration tests are located in [`/tests/integration`](https://github.com/xlab-si/xopera-opera/tree/master/tests/integration) directory. Both folders contain runnable TOSCA templates and playbooks, so there is no need to keep them separated. It would be nice if we could test also the TOSCA features that are present among examples but not in integration test which would allow us to be sure that we didn't break anything when adding new feature as opera integration tests are executed every time on push. Therefore I suggest moving opera's examples to integration tests folder.

## Steps
Implementation here should be straightforward. We would just move subfolders from `examples` to `tests/integration` folder and add `runme.sh` bash script so that every integration test could be runnable by CI/CD.

## Current behaviour
Right now opera keeps examples and integration tests separated.

## Expected results
To transfer all the examples in integration test folder.
______________
cc @sstanovnik this is what we talked about. However, we also agreed that we cannot move the examples that include any server establishing features (e.g. [OpenStack and nginx example](https://github.com/xlab-si/xopera-opera/tree/master/examples/nginx_openstack)) that you cannot easily test with CI/CD. So what should we do with these?

**Comment by sstanovnik** [comment]
While not optimal, I'd suggest putting the tests that can't be run automatically into a separate folder: `endtoend-examples`. Or similar.

These can still be considered tests, but executed manually with an appropriate external environment. Maybe a readme for each of them on their prerequisites and how to run them.

I'd also move everything from [xopera-examples](https://github.com/xlab-si/xopera-examples) here, as there isn't any sense in having two places for examples. Especially if #47 also removes the need for a separate docs repo.

**Comment by matejart** [comment]
There was a reasoning to the original schema. The `/examples` folder would contain simple, minimal deployment templates, which would be there to whet the user's appetite, prove to the user, that the tool works and is as easy as it says on the tin, show how basic things are done and not much more. The contents evolve with the features of the opera orchestrator. 

A separate repository is there to contain deployment templates for exploring deeper, elaborate use cases, which should be as independent from the life of the main repository as it is possible in the life of an experimental tool. If the `/examples` folder in this repository are about "how do I do a specific thing", then the separate repository is more about "what are all the wonderful options that TOSCA and opera opens up for me".

And the integration tests folder is purely for the purpose of development and testing and shouldn't even be meant to be looked at by people.

So, yes, there are in fact very good reasons to keep them apart. They are considered different deliverables with different target audiences. I'm afraid that the users will not notice examples meant for them unless they are clearly marked as `examples` or `docs/examples`. They will very likely not notice `endtoend-examples`. And moving them anywhere else will lose them entirely.

The aim to test that the examples still work in CI is noble and correct. However, examples are _not_ integration tests, so they need their own CI routine to test them.

**Comment by anzoman** [comment]
@matejart thanks for your thorough explanation of the schema. I agree, especially with the part where you presented that new opera users need some simple examples to start with without having to dig into the code or tests. On the other hand I thought and still think that the examples should be tested to as they now contain some TOSCA features that are not present among the integration tests. So as you have put it, we will probably new new CI/CD routine to test the examples. There is another possibility to keep the examples as they are now and to add any new features directly into integration tests but then the users could not experience the full power of opera since they would only see simple examples. And to be able to explore the advanced options of TOSCA they would have to move from here to [xopera-examples](https://github.com/xlab-si/xopera-examples) which, if decided to be kept, will also need some polishing and additions. 

I'm totally open to any better solution and further discussion about all this. As far as I get it you propose that we restructure the [linked PR](#94) in a way that we just add another CI routine that will test the examples.

@sstanovnik what do you think?

**Comment by anzoman** [comment]
İyi günler! Hi there, this issue has been death for a long time now. And the time has come for the awakening. We have figured out that moving the examples to integration test immediately would not be the best idea for the users. So, the PR #94 has been reorganized to establish a new CI routine to test the examples. @sstanovnik if you find some time, you can express your thoughts and take a look at the PR.

**Comment by anzoman** [comment]
Now we have a CI routine that tests the examples within this repository. Examples here are startup, simple and minimal TOSCA templates that will help users get started with `opera`. For more complex use cases (such as Docker, Kubernetes or cloud deployment) we kept and updated [xopera-examples](https://github.com/xlab-si/xopera-examples) repository. We will document all that so that users can understand our intentions better (connected PRs are here #171 and #172). The integration tests will be kept as they are now and will be improved constantly in order to test the most important opera features. I believe that this issue is not relevant anymore since we resolved all the dilemmas and we can therefore close it.
